### ISSUE TITLE
ndex in position 2 exceeds array bounds. Index must not exceed 1.

### ISSUE THREAD

**Issue by Frybank** [carnegie/PlantClusterFinder#7]
Dear author, I have encountered such problems recently, and I am looking forward to your guidance
**Index in position 2 exceeds array bounds. Index must not exceed 1.
Error in PlantClusterFinder>f_extract_results_with_header (line 4306)
Error in PlantClusterFinder (line 976)
MATLAB:badsubscript**

**Comment by Frybank** [comment]
GAP output file is not generated properly

**Comment by bxuecarnegie** [comment]
Line 976 corresponds to reading your gene location file, so there might be an issue with the format of said fild.

**Comment by Frybank** [comment]
> Line 976 corresponds to reading your gene location file, so there might be an issue with the format of said fild.
Does this mean a header format problem?
**Warning: 'Gene_Name_______Gene_Start__bp__Gene_End__bp____Chromosome_Name_Strand' exceeds MATLAB's maximum name length of 63 characters and has been truncated to 'Gene_Name_______Gene_Start__bp__Gene_End__bp____Chromosome_Name**

**Comment by bxuecarnegie** [comment]
Yes, use tab delimited files for your gene position files. 1 tab between each column.

**Comment by Frybank** [comment]
> Yes, use tab delimited files for your gene position files. 1 tab between each column.
I am using awk to grab the generated glof file, the previous test of successful data, also do so,According to the error message, I wonder if there is a problem with the watch head? My watch head is tab-separated
![微信图片_20231130145158](https://github.com/carnegie/PlantClusterFinder/assets/113606776/2c32343e-4d39-447f-a57c-2f414b03593b)

**Comment by bxuecarnegie** [comment]
I think there's definitely something with your header, if the lengths are correct, the script won't crop off "Strand", causing it to error since if cannot find the information. Try manually input the header and do it again.

**Comment by Frybank** [comment]
> I think there's definitely something with your header, if the lengths are correct, the script won't crop off "Strand", causing it to error since if cannot find the information. Try manually input the header and do it again.
Dear author, is there any of the following cases that the program runs completely and the output cluster has no data? The rubber and castor data I downloaded from ncbi, according to our process, did not get the output cluster

**Comment by Frybank** [comment]
> I think there's definitely something with your header, if the lengths are correct, the script won't crop off "Strand", causing it to error since if cannot find the information. Try manually input the header and do it again.
![微信图片_20231203124408](https://github.com/carnegie/PlantClusterFinder/assets/113606776/6782e74f-4661-412c-bcc9-2b62a26eccc9)
It seems that there is no N classification in the GAPOUT output, is there a problem with the data processing in this case?
![1](https://github.com/carnegie/PlantClusterFinder/assets/113606776/7692c363-582f-4168-a8bc-a17247f691b0)

**Comment by bxuecarnegie** [comment]
There are Ns in your screenshot. Please check if the scripts are missing N that are in your sequence or that your sequence lack gaps.
